### PR TITLE
Added managed memcache service and an AWS ElastiCache backend

### PR DESCRIPTION
**Pull request by tedsta**
This adds an abstraction over managed memcached services and implements an ElastiCache backend. It also adds a managed scenario to the memcached ycsb benchmark

TL;DR you can now run memcached ycsb against ElastiCache.

### CODE REVIEW

**Comment by tedsta** [comment]
@gareth-ferneyhough @yuyantingzero PTAL. Note this invovles small change to AwsFirewall

File: perfkitbenchmarker/memcache_service.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# Copyright 2016 PerfKitBenchmarker Authors. All rights reserved.
```
Comment (gareth-ferneyhough) [inline perfkitbenchmarker/memcache_service.py]: update to 2017

File: perfkitbenchmarker/providers/aws/elasticache.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
+# Copyright 2016 PerfKitBenchmarker Authors. All rights reserved.
```
Comment (gareth-ferneyhough) [inline perfkitbenchmarker/providers/aws/elasticache.py]: 2017

File: perfkitbenchmarker/providers/aws/elasticache.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    self.vpc_id = network.subnet.vpc_id
+    self.security_group_id = \
+        network.regional_network.vpc.default_security_group_id
+    self.subnet_id = network.subnet.id
+    self.subnet_group_name = '%ssubnet' % cluster_id
+
+  def Create(self):
+    # Open the port memcached needs
+    aws_network.AwsFirewall.GetFirewall() \
+        .AllowPortInSecurityGroup(self.region, self.security_group_id, 11211)
```
Comment (gareth-ferneyhough) [inline perfkitbenchmarker/providers/aws/elasticache.py]: Make port number (11211) a constant

File: perfkitbenchmarker/providers/aws/elasticache.py [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+        [(node['Endpoint']['Address'], node['Endpoint']['Port'])
+         for node in cluster_info['CacheNodes']]
+    assert len(self.hosts) == self.num_servers
+
+  def Destroy(self):
+    # Delete the ElastiCache cluster
+    cmd = ['aws', 'elasticache', 'delete-cache-cluster',
+           '--cache-cluster-id=%s' % self.cluster_id,
+           '--region=%s' % self.region]
+    out, _, _ = vm_util.IssueCommand(cmd)
```
Comment (gareth-ferneyhough) [inline perfkitbenchmarker/providers/aws/elasticache.py]: If you aren't using `out`, do you need to save it and its underscore friends?

File: perfkitbenchmarker/providers/gcp/memcache.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
+# Copyright 2016 PerfKitBenchmarker Authors. All rights reserved.
```
Comment (gareth-ferneyhough) [inline perfkitbenchmarker/providers/gcp/memcache.py]: 2017

**Comment by gareth-ferneyhough** [comment]
LGTM

**Pull request merged**